### PR TITLE
Add DragComponent tests

### PR DESCRIPTION
**Pull request by TrySound**
- fixed `for` type
- fixed DragComponent `isOverAccepted` state
- fixed default Draggable `type` which previously produced `isOverAccepted: true` when dragging cursor is not over `Droppable` and its `accepts` prop is string
- changed `cjs` target to `umd` which adds not significant runtime, but allows to use result directly in browser which I used for puppeteer tests
- fixed `isOverAccepted` when Droppable `accepts` is array 

Each test is quite massive, but descriptive and produces more real world behavior than events emulating.

### CODE REVIEW

**Comment by TrySound** [comment]
Tests are important :)

File: src/Draggable.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ class Draggable extends React.Component {
     onDragStart: () => {},
     onDrag: () => {},
     data: null,
-    type: null,
+    type: '',
```
Comment (chrisjpatty) [inline src/Draggable.js]: I don't know about this change here, I think it makes more sense to send a a null if nothing is being draggged. It would be more consistent with the other properties of the state.
Is this change intended to add better support for flow by not switching between null and a string?

Comment (TrySound) [reply src/Draggable.js]: Nope, it's about behavior.
> fixed default Draggable type which previously produced isOverAccepted: true when dragging cursor is not over Droppable and its accepts prop is string

Comment (TrySound) [reply src/Draggable.js]: We need some more valuable default than `null`. Or we can make it required instead

Comment (TrySound) [reply src/Draggable.js]: Also we can make `accepts` always array then `null` won't be found in it.

Comment (TrySound) [reply src/Draggable.js]: I got an idea. We can you array as `accepts` default instead of `null`. It shouldn't be a breaking change.

**Comment by TrySound** [comment]
Are you okay with other changes?

**Comment by chrisjpatty** [comment]
I haven't had a chance yet to fully review them all, but the tests look great! Thanks for writing those, I've been dreading having to sit down and write them haha

File: src/Draggable.js (hunk 1 above)
Comment (chrisjpatty) [reply src/Draggable.js]: I hope I'm understanding correctly. I think it would make the most sense to have `isOverAccepted` evaluate to `true` if the `accepts` prop of the droppable is either null or undefined. That way you can default to accepting everything. The docs should probably be updated to make that clear.

Comment (TrySound) [reply src/Draggable.js]: But the problem is that `isOverAccepted: true` even before hovering droppable. It's just always true.

Comment (chrisjpatty) [reply src/Draggable.js]: Oh I see what you're saying. So it evaluates to true if they're both null right? I guess in that case we could a truthy check on the currently hovered droppable id like:
```
isOverAccepted: currentlyHoveredDroppableId
    ? Array.isArray(this.currentlyHoveredDroppableAccepts)
        ? this.currentlyHoveredDroppableAccepts.includes(state.type)
            : state.type === this.currentlyHoveredDroppableAccepts
        : false
```

Comment (chrisjpatty) [reply src/Draggable.js]: Would that resolve what you're seeing?

Comment (TrySound) [reply src/Draggable.js]: I will add test case for default

**Comment by TrySound** [comment]
I added internal name to distinct with uninitialized value

**Comment by chrisjpatty** [comment]
When I pulled down the most recent changes I'm getting some unexpected behavior on the droppable now. The DragComponent is getting that const value you've added to the store as the currentlyHoveredDroppableAccepts instead of a null as expected. https://www.useloom.com/share/055d3d10809d4446916be0e0eae39986

**Comment by TrySound** [comment]
@chrisjpatty It's expected. Special value which is used as default type and distincted from not initialized value which is `null`. You can play with different implementations, but current tests should always pass.

**Comment by chrisjpatty** [comment]
The currentlyHoveredDroppableAccepts is intended to be a Boolean representing if it will accept it, not what it accepts. It would be unexpected to get a string on that key. I misspoke in my last comment saying it should expect a null. The developer using this package shouldn't need to know that the string constant represents an uninitialized value.

**Comment by chrisjpatty** [comment]
I think the uninitialized value and the default value for the accepts props should probably both be the same. I'm concerned that an arbitrary string value as the default would be confusing to the developer using the package..

**Comment by chrisjpatty** [comment]
*I deleted one of my last comments because I misspoke. I was thinking of the wrong prop.

**Comment by TrySound** [comment]
@chrisjpatty Well, they can't be the same since behavior becomes wrong. The value becomes meaningful only when mouse over droppable. But if default and uninitialized value are the same and you get the same. The case is not about types, but the correct value of `isOverAccepted`. internal name is okay for developer, I think. Alternative is preventing default at all.

**Comment by chrisjpatty** [comment]
I hope I'm understanding what you're getting at correctly. In my mind if no accepts prop is provided it should default to `null`, and if no type is provided it should also default to `null`. Then when the DragComponent compares the currentlyHoveredDroppableAccepts to the type, it it should first check if the accepts is null. If it is, it means no accepts was provided so it should accept everything and return true, but if you're not over a droppable, then you know that it should always return false. Like this:
```jsx
render() {
		const state = store.getState()
		const accepts = state.currentlyHoveredDroppableAccepts
		const isOverDroppable =
			typeof state.currentlyHoveredDroppableId === 'string'
		return (
			state.isDragging &&
			state.currentlyDraggingId === this.props.for &&
			this.props.children({
				...state,
				isOverAccepted: isOverDroppable
					? accepts !== null
						? Array.isArray(accepts)
							? accepts.find(state.type)
							: state.type === accepts
						: true
					: false
			})
		)
	}
```

**Comment by TrySound** [comment]
Not `find`, but `includes`

**Comment by chrisjpatty** [comment]
Oh right, that is the right method in this case. Does that resolve the issue you're seeing?

**Comment by TrySound** [comment]
Yeah, that works. However still a bit complicated. But nevermind. We can reconsider it later. The most important is passed tests.

**Comment by chrisjpatty** [comment]
Yeah I'm totally willing to revisit this later. Thanks again for another great pull request!

**Comment by TrySound** [comment]
Let's make minor release with flow types and fixes

**Comment by chrisjpatty** [comment]
Yeah sure I'll push one out.
